Sources/Common/Module/ModuleTopLevelObject.swift

```diff
@@ -7,33 +7,49 @@ import Foundation
 // Top-level class meaning it contains public facing SDK functions called by customers.
 // There isn't a constructor populated via dependency injection. It's at the top node
 // of dependencies.
-// TODO: [CDP] Remove class if no longer needed
 open class ModuleTopLevelObject<ImplementationClass> {
     private(set) var alreadyCreatedImplementation: ImplementationClass?
     public var implementation: ImplementationClass? {
-        alreadyCreatedImplementation ?? createAndSetImplementationInstance()
+        let instance = alreadyCreatedImplementation ?? createAndSetImplementationInstance()
+        if instance == nil {
+            logger.info("Module \(moduleName) is not yet initialized")
+        }
+        return instance
```

Comment:
This ensures safe access to an already initialized instance of the module and generates a log message if any method from the module is accessed before initialization.